core/trino-main/src/main/java/io/trino/server/protocol/JsonEncodingUtils.java

```diff
@@ -143,10 +160,51 @@ public static void writePagesToJsonGenerator(Consumer<TrinoException> throwableC
         }
     }
 
+    private static String describeBlock(Block block)
+    {
+        return switch (block) {
+            case DictionaryBlock dictionaryBlock -> "DictionaryBlock{%s}".formatted(describeBlock(dictionaryBlock.getDictionary()));
+            case RunLengthEncodedBlock runLengthEncodedBlock -> "RleBlock{%s}".formatted(runLengthEncodedBlock.getValue());
+            case ArrayBlock arrayBlock -> "ArrayBlock{%s}".formatted(describeBlock(arrayBlock.getElementsBlock()));
+            case MapBlock mapBlock -> "MapBlock{keys=%s, values=%s}".formatted(describeBlock(mapBlock.getKeyBlock()), describeBlock(mapBlock.getValueBlock()));
+            case RowBlock rowBlock -> "RowBlock{fields=%s}".formatted(rowBlock.getFieldBlocks().stream().map(JsonEncodingUtils::describeBlock).collect(joining(", ")));
+            default -> block.toString();
```

Comment:
This code should not be here for ever.
Let's file an issue describing problem we're trying to identify and add a TODO comment + link here to eventually remove this boiler-plate.
Also, it would probably be enough to handle primitive types only. Code would be much simpler, while debuggability would be near exact the same.